cassiopeia/simulator/CellSubsampler.py

```diff
@@ -0,0 +1,124 @@
+import abc
+import networkx as nx
+import numpy as np
+from typing import Optional
+
+from cassiopeia.data import CassiopeiaTree
+
+
+class CellSubsamplerError(Exception):
+    """An Exception class for the CellSubsampler class."""
+
+    pass
+
+
+class CellSubsampler(abc.ABC):
+    r"""
+    Abstract base class for all cell samplers.
+
+    A CellSubsampler implements a method 'subsample' which, given a Tree,
+    returns a second Tree which is the result of subsampling cells
+    (i.e. leafs) of the tree. Only the tree topology will be created for the
+    new tree.
+    """
+
+    @abc.abstractmethod
+    def subsample(self, tree: CassiopeiaTree) -> CassiopeiaTree:
+        r"""
+        Returns a new CassiopeiaTree which is the result of subsampling
+        the cells in the original CassiopeiaTree.
+
+        Args:
+            tree: The tree for which to subsample leaves.
+        """
+
+
+class UniformCellSubsampler(CellSubsampler):
```

Comment:
Should we make a new file for this class? I presume we'll have different CellSubsamplers?

Because if not, we should just make this part of the `CassiopeiaTree` class I think. (In the form of having a method dubbed `subsample` that takes in some fraction and randomly downsamples the tree)